relayer/chainproxy/tendermintRPC.go

```diff
@@ -130,7 +134,19 @@ func (cp *tendermintRpcChainProxy) FetchBlockHashByNum(ctx context.Context, bloc
 	return hash, nil
 }
 
-func NewtendermintRpcChainProxy(nodeUrl string, nConns uint, sentry *sentry.Sentry, csm *lavasession.ConsumerSessionManager, pLogs *PortalLogs) ChainProxy {
+func NewtendermintRpcChainProxy(nodeUrl string, nConns uint, sentry *sentry.Sentry, csm *lavasession.ConsumerSessionManager, pLogs *PortalLogs, flagSet *pflag.FlagSet) ChainProxy {
+	var httpUrl string
+	if nodeUrl != "" { // provider process
+		var err error
+		httpUrl, err = flagSet.GetString(rpcprovider.TendermintProviderHttpEndpoint)
```

Comment:
we need to verify the other is websocket, and that this is http, and do it in protocol/chainlib/chainproxy